test/test_images/eventdetails/main.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+
+	cloudevents "github.com/cloudevents/sdk-go"
+)
+
+func getHeader(header http.Header, headerName string) *string {
+	if header != nil && len(header) > 0 {
+		for k, v := range header {
+			if k == headerName {
+				return &v[0]
+			}
+		}
+	}
+	return nil
+}
+
+//func handler(event cloudevents.Event) {
+func handler(ctx context.Context, event cloudevents.Event) {
+	fmt.Printf("Got Event Context: %+v\n", event.Context)
+	tx := cloudevents.HTTPTransportContextFrom(ctx)
+	fmt.Printf("Got Transport Context: %+v\n", tx)
+	fmt.Printf("----------------------------\n")
+	header := tx.Header
+	headerNameList := []string{"X-B3-Sampled", "X-B3-Traceid", "X-B3-Spanid", "X-B3-ParentSpanId", "X-Custom-Header"}
+	for _, headerName := range headerNameList {
+		headerValue := getHeader(header, headerName)
```

Comment:
```golang
if headerValue := header.Get(headerName); headerValue != "" {
    fmt.Printf("Got Header %s: %s\n", headerName, *headerValue)
} else {
    fmt.Printf("Missing Header %s\n", headerName)
}
```

Reply:
changed

```go
		if headerValue := getHeader(header, headerName); headerValue != nil {
			fmt.Printf("Got Header %s: %s\n", headerName, *headerValue)
		} else {
			fmt.Printf("Missing Header %s\n", headerName)
		}
```

Reply:
I don't think this was done.

Reply:
This comment is mostly around the fact that `getHeader` is not needed, and actually we would prefer to use `header.Get`.

Reply:
I see now - I missed the point. Can header value be empty string and be distinguished from to lack of header? I changed to use  header.Get

Reply:
You are right, with this interface we cannot distinguish empty string vs does not exist. However, for testing of tracing flags, is there a meaningful difference?

Reply:
I think it should not matter - headers would be equivalent of missing header?